apps/site/lib/site_web/controllers/schedule/map_api.ex

```diff
@@ -4,31 +4,35 @@ defmodule SiteWeb.ScheduleController.MapApi do
   """
   use SiteWeb, :controller
 
+  alias Leaflet.MapData
+  alias Routes.Route
   alias SiteWeb.ScheduleController.Line.Helpers, as: LineHelpers
   alias SiteWeb.ScheduleController.Line.Maps
   alias Stops.Repo, as: StopsRepo
+  alias Stops.Stop
 
-  @type query_param :: String.t() | nil
   @type direction_id :: 0 | 1
 
+  @spec show(Plug.Conn.t(), map()) :: Plug.Conn.t()
   def show(conn, %{
         "id" => route_id,
         "direction_id" => direction_id
       }) do
     map_data =
-      get_map_data(conn, route_id, String.to_integer(direction_id), %{
-        stops_by_route_fn: &StopsRepo.by_route/3
-      })
+      get_map_data(conn, route_id, String.to_integer(direction_id), &StopsRepo.by_route/3)
 
     json(conn, map_data)
   end
 
-  def get_map_data(conn, route_id, direction_id, deps) do
+  @type stops_by_route_fn :: (Route.id_t(), 0 | 1, Keyword.t() -> [Stop.t()] | {:error, any})
+  @spec get_map_data(Plug.Conn.t(), Route.id_t(), direction_id(), stops_by_route_fn()) ::
+          MapData.t()
+  defp get_map_data(conn, route_id, direction_id, stops_by_route_fn) do
     route = LineHelpers.get_route(route_id)
-    variant = conn.query_params["variant"]
-    route_shapes = LineHelpers.get_route_shapes(route.id, direction_id)
-    route_stops = LineHelpers.get_route_stops(route.id, direction_id, deps.stops_by_route_fn)
-    active_shapes = LineHelpers.get_active_shapes(route_shapes, route, variant)
+    shape_id = conn.query_params["shape_id"]
+    route_shapes = LineHelpers.get_route_shapes(route.id, direction_id, false)
```

Comment:
Thank you for finding these! I made this change:

> Revert the "Don't filter out shapes with negative priority." change
> 
> This let through a lot of shapes we don't want to display. I think we actually need to be filtering based on route patterns instead of shape priority (which is deprecated), but I'll create a different ticket for that.

It turns out the filtering thing was trying to deal with a separate issue than this ticket describes. The 66 was just a particularly interesting example because it demonstrated _both_. So with this change, you won't actually see the correct map for this variant of the 66, but looking at the 111 you can see the map now toggles properly where it doesn't in production.

I'll make a new ticket for the variant issue. That 66 variant isn't even running right now, and I think it's an open question about whether we should show it when it is.

Reply:
Ticket for the variant issue: https://app.asana.com/0/385363666817452/1179129122227738